sentry-android-core/src/main/java/io/sentry/android/core/AndroidContinuousProfiler.java

```diff
@@ -88,12 +90,14 @@ private void init() {
             buildInfoProvider);
   }
 
-  public synchronized void setScopes(final @NotNull IScopes scopes) {
-    this.scopes = scopes;
-    this.performanceCollector = scopes.getOptions().getTransactionPerformanceCollector();
-  }
-
   public synchronized void start() {
+    if ((scopes == null || scopes != NoOpScopes.getInstance())
+        && Sentry.getCurrentScopes() != NoOpScopes.getInstance()) {
+      this.scopes = Sentry.getCurrentScopes();
```

Comment:
in theory you could pass `Scopes` as an argument here, because this method is called from within `Scopes`

Reply:
it is also called in the `SentryPerformanceProvider`, when it's used as app start profiler. That's why i'm retrieving the scopes whenever I can